webapp/src/index.tsx

```diff
@@ -211,21 +204,30 @@ export default class Plugin {
         });
 
         registry.registerWebSocketEventHandler(`custom_${pluginId}_call_start`, (ev) => {
+            const channelID = ev.broadcast.channel_id;
+
             store.dispatch({
                 type: VOICE_CHANNEL_CALL_START,
                 data: {
-                    channelID: ev.broadcast.channel_id,
+                    channelID,
                     startAt: ev.data.start_at,
                     ownerID: ev.data.owner_id,
                 },
             });
             store.dispatch({
                 type: VOICE_CHANNEL_ROOT_POST,
                 data: {
-                    channelID: ev.broadcast.channel_id,
+                    channelID,
                     rootPost: ev.data.thread_id,
                 },
             });
+
+            if (window.callsClient?.channelID === channelID) {
```

Comment:
Does this mean the `callsClient` has to be running before the `_call_start` event for the thread to be followed? Is that a possible race condition?

Reply:
If we are the user that started the call then it's more or less guaranteed as it means we already initialized the client to send the join message. The only weird case that could happen is:

- Init client
- Start a call
- Something fails and we drop
- Deinit client
- Start event is received, nothing happens

But again, if we were to join again (assuming the call is still on) we would follow anyway so I think it's okay.

Reply:
What if we're a user joining the call? (I think it would work, but I just want to double check I'm following the order of things.)

Reply:
If we are joining the call the call we fallback to the other cases which will just fetch the thread id from the store and follow the thread directly. The problem we were originally addressing here was that upon starting a call the thread id wasn't in the store yet so we moved the call to the call_start event handler.

Reply:
Ah, I see now, my mistake. I didn't read carefully enough. 👍